src/providers/google/youtube-favourite.ts

```diff
@@ -0,0 +1,184 @@
+import GoogleHandler from "./GoogleHandler";
+import CONFIG from "../../config";
+import { SyncProviderLogEvent, SyncProviderLogLevel } from '../../interfaces'
+import {
+    SyncResponse,
+    SyncHandlerPosition,
+    SyncHandlerStatus,
+} from "../../interfaces";
+import { SchemaFavouriteContentType, SchemaFavouriteType, SchemaFavourite } from "../../schemas";
+import { google, youtube_v3 } from "googleapis";
+import { GaxiosResponse } from "gaxios";
+
+const _ = require("lodash");
+
+export default class YouTubeFavourite extends GoogleHandler {
+
+    public getName(): string {
+        return "youtube-favourite";
+    }
+
+    public getSchemaUri(): string {
+        return CONFIG.verida.schemas.FAVOURITE;
+    }
+
+    public getProviderApplicationUrl(): string {
+        return "https://youtube.com/";
+    }
+
+    public getYouTube(): youtube_v3.Youtube {
+        const oAuth2Client = this.getGoogleAuth()
+        const youtube = google.youtube({ version: "v3", auth: oAuth2Client });
+        return youtube;
+    }
+
+    public async _sync(
+        api: any,
+        syncPosition: SyncHandlerPosition
+    ): Promise<SyncResponse> {
+        const youtube = this.getYouTube();
+    
+        const query: youtube_v3.Params$Resource$Videos$List = {
+            part: ["snippet", "contentDetails"],
+            myRating: "like",
+            maxResults: this.config.batchSize, // Google Docs: default = 5, max = 50
+        };
+    
+        if (syncPosition.thisRef) {
+            query.pageToken = syncPosition.thisRef;
+        }
+    
+        const serverResponse = await youtube.videos.list(query);
+    
+        if (
+            !_.has(serverResponse, "data.items") ||
+            !serverResponse.data.items.length
+        ) {
+            // No results found, so stop sync
+            syncPosition = this.stopSync(syncPosition);
+    
+            return {
+                position: syncPosition,
+                results: [],
+            };
+        }
+        
+        // Sort items by publishedAt timestamp in descending order (most recent first)
```

Comment:
What's the purpose of this code?

It's only sorting locally.

Same question applies to the other youtube handlers.

Reply:
Yeah, it is for local sort.
Do you want to remove them? @tahpot

Reply:
I removed the sort.